arduino/smartcar/smartcar.ino

```diff
@@ -19,48 +19,67 @@ void setup()
 
 void loop()
 {
-  handleInput();
-  delay(35);
+    handleInput();
+    delay(35);
 }
 
 void handleInput()
 {
-  float distance = front.getDistance();
-  Serial.println(distance);
-  if (Serial.available())                           //If the user enters input in the serial The car must move according to that input.
-  {
-   String input = Serial.readStringUntil('\n');
-   if (input.startsWith("m"))     
-   {
-     int cSpeed = input.substring(1).toInt();
-     magnitude = cSpeed;                           //We save the user's input in here in order to have it outside of the if scope.                     
-     car.setSpeed(cSpeed); 
-   } else if (input.startsWith("t"))
-   {
-     int cAngle = input.substring(1).toInt();
-     car.setAngle(cAngle);
-     delay(600);                                                      //This delay is needed for the car to turn in a short while and then go back to its straight direction,  
-   }                                                                  // because we dont want the car to to turn around itself for no reason!
-   if (distance > 1 && distance < 200)          //This is the obstacle treshhold. The car must detect obstacle closer than 200
-   {                                                                  // and avoid them
-     handleObstacle();
-   }  
-   car.setSpeed(magnitude);                                           //this makes sure the car is back to its forward direction if a turning happened.
-   car.setAngle(0);                                                   //and this!
-  } else if (!Serial.available())                                     //This happens if there is no input from user in the serial.
-  {
-    if (distance > 1 && distance < 200)   
+    float distance = front.getDistance();
+    serialMsg(distance);
+    if (Serial.available())            //If the user enters input in the serial The car must move according to that input.
     {
-      handleObstacle();
+        String input = Serial.readStringUntil('\n');
+        serialReader(input);
+        distanceHandler(0, 200, distance);
+    } else if (!Serial.available()) {
```

Comment:
Wouldn't a simple `else` have worked in this case since you are handling the same variable?